001-project-file-structure/src/app/components/directives/card-generator/card-generator.directive.js

```diff
@@ -0,0 +1,38 @@
+(function() {
+  'use strict';
+
+  angular
+    .module('angularEx1')
+    .directive('cardGenerator', cardGenerator);
+
+  /** @ngInject */
+  function cardGenerator() {
+    return {
+      restrict: 'E',
+      replace: true,
+      scope: {
+        icon: '=',
+        textColor: '=',
+        backgroundColor:'=',
+        title: '=',
+        descriptionText: '=',
+        favorite: '&'
+
+      },
+      templateUrl: 'app/components/directives/card-generator/card-generator.html',
+      link: function ($scope) {
+        $scope.markAsFavorite = function() {
```

Comment:
if you work with a whole object you can avoid re creating another object every time it is set to favorite and avoid the use of the link function, but this way also works.